tests/core/test_nlg.py

```diff
@@ -328,3 +328,126 @@ def test_nlg_fill_response_quick_replies(
         filled_slots={quick_replies_slot_name: quick_replies_slot_value},
     )
     assert result == {"quick_replies": str(quick_replies_slot_value)}
+
+
+def test_nlg_conditional_response_variations():
+    responses = {
+        "utter_test": [
+            {
+                "text": "Conditional Response A",
+                "condition": [{"type": "slot", "name": "slot test", "value": "A"}],
+                "channel": "os",
+            },
+            {
+                "text": "Conditional app response A",
+                "condition": [{"type": "slot", "name": "slot test", "value": "A"}],
+                "channel": "app",
+            },
+            {
+                "text": "Conditional Response B",
+                "condition": [{"type": "slot", "name": "slot test", "value": "B"}],
+            },
+            {"text": "Default response"},
+        ]
+    }
+    t = TemplatedNaturalLanguageGenerator(responses=responses)
+
+    condition_a = t._matches_filled_slots(
+        filled_slots={"slot test": "A"}, response=responses["utter_test"][0]
+    )
+    assert condition_a is True
+
+    condition_b = t._matches_filled_slots(
```

Comment:
this seems like multiple separate tests to me. I'd split this either in different tests or use `pytest.parametrize`  (e.g. see `Unit Tests Should Verify a Single-Use Case` [here](https://dzone.com/articles/unit-testing-best-practices-how-to-get-the-most-ou)

Reply:
e.g. named tests would make it easier to me to grasp which case which assertion is testing

Reply:
(same for the other tests btw)